### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by owais-vd**
Bugs found in comments of issue #19

### CODE REVIEW

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 1]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
         }
         context.startActivity(intent);
     }
+
+    public static LocalDate getNextContactVisitDate(JSONObject jsonObject) {
+        String methodExit = getJsonFieldValue(jsonObject, "step7", "method_exit");
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded strings to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 2]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+
+        DateTimeFormatter pattern = DateTimeFormat.forPattern(("dd-MM-yyyy"));
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded date format string to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 3]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+        LocalDate nextContactDate = null;
+        switch (methodExit) {
+            case "cu_iud":
+            case "lng_iud":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "iud_insertion_date"), pattern);
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded strings to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 4]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+                break;
+
+            case "dmpa_im":
+            case "dmpa_sc":
+            case "net_en":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "last_injection_date"), pattern);
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded strings to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 5]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+                break;
+
+            case "pop":
+            case "coc":
+            case "patch":
+            case "cvr":
+            case "pvr":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step1", "visit_date"), pattern);
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded strings to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 6]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+                break;
+
+            case "male_sterilization":
+            case "female_sterilization":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "sterilization_date"), pattern);
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded strings to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 7]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+        }
+
+        return nextContactDate;
+    }
+
+    private static String getJsonFieldValue(JSONObject jsonObject, String stepKey, String fieldKey) {
+        String value = "";
+
+        try {
+            JSONArray jsonArray = jsonObject.getJSONObject(stepKey).getJSONArray("fields");
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded string to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java [hunk 8]
```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
...
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject obj = jsonArray.getJSONObject(i);
+                if (obj.getString("key").equals(fieldKey)) {
+                    value = obj.optString("value", "");
```
Comment (rkareko) [inline opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Please move the hard coded String to a constants file

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 1 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 2 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 3 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 4 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 5 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 6 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 7 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

File: opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java (hunk 8 above)
Comment (owais-vd) [reply opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java]: Done

**Review by Rkareko — APPROVED** [review]

**Pull request merged**